Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessing.cs

```diff
@@ -52,154 +51,86 @@ internal ProfilerSqlProcessing(TelemetryConfiguration configuration, string agen
-        {
-            return this.OnBegin(thisObj, false);
-        }
-
-        /// <summary>
-        /// On begin callback for sync methods except ExecuteReader.
-        /// </summary>
-        /// <param name="thisObj">This object.</param>        
-        /// <returns>The context for end callback.</returns>
-        public object OnBeginForSync(object thisObj)
+        public object OnBeginForOneParameter(object thisObj)
         {
-            return this.OnBegin(thisObj, false);
+            return this.OnBegin(thisObj);
         }
 
         /// <summary>
-        /// On end callback for ExecuteReader.
+        /// On begin callback for methods with 2 parameter.
         /// </summary>
-        /// <param name="context">The context.</param>
-        /// <param name="returnValue">The return value.</param>
-        /// <param name="thisObj">This object.</param>
-        /// <param name="behavior">The callback parameter.</param>
-        /// <param name="method">The state parameter.</param>
-        /// <returns>The resulting return value.</returns>
-        public object OnEndForExecuteReader(object context, object returnValue, object thisObj, object behavior, object method)
+        public object OnBeginForTwoParameters(object thisObj, object parameter1)
         {
-            this.OnEnd(context, null, thisObj, false);
-            return returnValue;
+            return this.OnBegin(thisObj);
         }
 
         /// <summary>
-        /// On end for sync methods except ExecuteReader callback.
+        /// On begin callback for methods with 3 parameters.
         /// </summary>
-        /// <param name="context">The context.</param>
-        /// <param name="returnValue">The return value.</param>
-        /// <param name="thisObj">This object.</param>        
-        /// <returns>The resulting return value.</returns>
-        public object OnEndForSync(object context, object returnValue, object thisObj)
+        public object OnBeginForThreeParameters(object thisObj, object parameter1, object parameter2)
         {
-            this.OnEnd(context, null, thisObj, false);
-            return returnValue;
+            return this.OnBegin(thisObj);
         }
 
         /// <summary>
-        /// On exception callback for ExecuteReader.
+        /// On begin callback for methods with 4 parameter.
         /// </summary>
-        /// <param name="context">The context.</param>
-        /// <param name="exception">The exception.</param>
-        /// <param name="thisObj">This object.</param>
-        /// <param name="behavior">The callback parameter.</param>
-        /// <param name="method">The state parameter.</param>
-        public void OnExceptionForExecuteReader(object context, object exception, object thisObj, object behavior, object method)
+        public object OnBeginForFourParameters(object thisObj, object parameter1, object parameter2, object parameter3)
```

Comment:
Stylecop does not complain about missing parameters if they all miss. Xml documentation would be useless in this case because it is actually param1,2,3 etc The meaning will vary depending on when method you instrument. Here we can write only something like "First argument of callback method. Second argument ..." but stylecop will complain that descriptions differ only by one word.

---
In reply to: [82120587](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/170#discussion_r82120587) [](ancestors = 82120587)